### PR TITLE
Return empty string for missing response bodies

### PR DESCRIPTION
**Pull request by keathley**
So, after using starting to roll this out into production I've received a bunch of feedback that we should default the response body to an empty string. After thinking about it more I think that is probably the correct choice for a few reasons:

* It's not "wrong" semantically speaking. This is the most important point IMO. From the point of view of the parser, we'd expect it to return something like an empty string.
* It removes additional error checking logic that users need to do.
* There's a precedent for this in elixir already. The other clients return empty strings.
* Clients in other languages also use an empty string. I had only glanced at the ruby client. At this point, I'm pretty sure ruby is the outlier here.

That's my reasoning for making this change now.

TL;DR - @QuinnWilton was right :heart:

### CODE REVIEW

File: lib/finch/response.ex [hunk 1]
```diff
@@ -7,14 +7,14 @@ defmodule Finch.Response do
...
-    :body,
+    body: "",
     headers: []
   ]
 
   @typedoc """
   A body associated with a response.
   """
-  @type body() :: binary() | nil
+  @type body() :: binary()
```
Comment (wojtekmach) [inline lib/finch/response.ex:17]: consider just inlining this type: `%Response{..., body: binary(), ...}`. perhaps it's useful to export it, idk, so please feel free to just ignore me. :)
